fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryWrapper.java

```diff
@@ -159,6 +169,8 @@ public IndexWriter getWriter(LuceneAnalyzerWrapper analyzerWrapper) throws IOExc
                     }
                     writer = new IndexWriter(directory, indexWriterConfig);
                     writerAnalyzerId = analyzerWrapper.getUniqueIdentifier();
+                    // Merge is required when creating an index writer (do we have a better indicator for a required merge?)
```

Comment:
This seems excessive. but if we can't get something better from Lucene itself, it could be any time we do any write or delete, which is when we create the IndexWriter. It's possible that the index maintainer should reuse the `IndexWriter` for more operations.

Reply:
This is what we have today. The overhead would be re-reading the index information (and the means required to do a task in the background, if the record update caller chooses to go this way).